Catch MQTT payload encoding errros

hartmms opened the pull request:
**Description:**

If a MQTT payload is not utf-8 encoded data, the MQTT component will try to decode it as such regardless and throw an exception.  This exception kills the component, but leaves the frontend running.  

This change will trap for this exception and generate a logfile error message instead. 

**Related issue (if applicable):** fixes #

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#

**Example entry for `configuration.yaml` (if applicable):**

``` yaml

```

**Checklist:**

If user exposed functionality or configuration variables are added/changed:
- [ ] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If code communicates with devices, web services, or a:
- [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
- [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L16)).
- [ ] New dependencies are only imported inside functions that use them ([example](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L51)).
- [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
- [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
- [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
- [ ] Tests have been added to verify that the new code works.

Commit:
customize if pushetta should send test message on start up

hartmms committed:
added error handling to mqtt message receive if payload is not utf-8 unicode
added mqtt test for above code as well

pvizeli committed:
Big Homematic update (#3677)

* Update homeassistant with new pyhomematic layout

* fix linter

fabaff reviewed (changes requested):
Your PR will change the file permission from 644 to 755.

hartmms committed:
change permission back to 644

hartmms commented:
Not sure why permissions got messed up.  should be fixed now.

robbiet480 committed:
Add @mention-bot configuration

balloob committed:
Address asyncio comments (#3663)

* Template platforms: create_task instead of yield from

* Automation: less yielding, more create_tasking

* Helpers.script: less yielding, more create_tasking

* Deflake logbook test

* Deflake automation reload config test

* MQTT: Use async_add_job and threaded_listener_factory

* Deflake other logbook test

* lint

* Add test for automation trigger service

* MQTT client can be called from within async

balloob commented:
Original PR #3326

Commit:
Customize initial state of automation

balloob commented:
My comment still has to be resolved: https://github.com/home-assistant/home-assistant/pull/3326#issuecomment-246569974

balloob commented on tests/components/mqtt/test_init.py:
```diff
@@ -118,6 +118,20 @@ def test_service_call_with_payload_doesnt_render_template(self):
         }, blocking=True)
         self.assertFalse(mqtt.MQTT_CLIENT.publish.called)
 
+    def test_service_call_without_payload_or_payload_template(self):
```
No need to add this test back. It cannot happen because of voluptuous.

hartmms replied:
I'm confused. Are you saying no test is required for the new code I added to components/mqtt/__init__.py or are you saying the way I tested it is wrong and needs to conform to what you prototyped out in the previous push request?

balloob replied:
This comment is attached to a line. You have added a removed test back which was removed because it was impossible to happen.

hartmms replied:
doh. stupid me realized what I goofed up.  The "old" test is removed and I added the proper "new" test for the code I changed.

fabaff committed:
Upgrade Sphinx to 1.4.8 and set missing variables (#3650)

deisi committed:
Fix 3621 (#3642)

Happens when the scanner searches for a mac address, that has been forgotten
by the fritzbox.

Commit:
Report availability of arest (#3614)

molobrakos committed:
Support for encrypted payload (#3587)

molobrakos committed:
only query vehicle attributes once (#3668)

use registration number as dev id

capellini committed:
test: mocking IO in HTML5 notify tests (#3685)

Replacing temporary file creation in tests with mock's mock_open for faster IO.

fabaff committed:
Statistics sensor (#3513)

* Initial stats sensor

* Add total and tests

* Use deque, rename var, set default, and only update sensor

fabaff committed:
Auth and headers support for REST sensor (#3592)

* Add auth and header support

* Update header part

Commit:
Added support for Wink Smoke and CO detectors (#3645)

sam-io committed:
fix for date formatting issue (#3693)

persandstrom committed:
vsure 0.10.3

robbiet480 committed:
Update .mention-bot to add more users to blacklist and remove skipAlreadyMentionedPR

robbiet480 committed:
Now no one wants to be blacklisted, so lets remove the configuration entirely

balloob committed:
Add async_safe annotation (#3688)

* Add async_safe annotation

* More async_run_job

* coroutine -> async_save

* Lint

* Rename async_safe -> callback

* Add tests to core for different job types

* Add one more test with different type of callbacks

* Fix typing signature for callback methods

* Fix callback service executed method

* Fix method signatures for callback

pvizeli committed:
Service & signal (stop/restart) fix (#3690)

* Bugfix signhandling/services

* change from coroutine to callback

* add error handling

* fix bug with endless running

* fix unit test

* Revert "fix unit test"

This reverts commit 31135c770923161f7afb3a31f4dd4fea99533a9c.

* Disable sigterm/sighup test

Danielhiversen committed:
Add automations and scripts to group.all_automations and group.all_scripts (#3664)

* Add automations to group.all_automations

* Add scripts to group.all_scripts

gwendalg committed:
Add Ted5000 Sensor (#3559)

* Add Ted5000 Sensor

Signed-off-by: Gwendal Grignou <gwendal@gmail.com>

* sensors: ted5000: use requests instead of urllib.

Signed-off-by: Gwendal Grignou <gwendal@gmail.com>

* Update ted5000.py

Add timeout to requests call.

turbokongen committed:
Add set_config_parameter service (#3696)

pvizeli committed:
Update ha-alpr / change default interval (#3691)

fabaff committed:
Check that no configuration is provided (#3675)

balloob committed:
Fix broken tests (#3704)

* Fix broken tests

* Lint

tchellomello committed:
Fixed issue #3624 and bumped python-nest to 2.10.0 version (#3665)

* Fixed issue #3624 and bumped python-nest to 2.10.0 version

*    Fixed return state when accessing attribute operation_mode

* Fixed typo

persandstrom committed:
Add västtrafik sensor (#3640)

MartinHjelmare committed:
Set force_update to true for mysensors sensors (#3648)

balloob committed:
Speed up MQTT server test (#3703)

balloob committed:
Update frontend

turbokongen committed:
Add possibility to set temperature to a given operation_mode (#3646)

* Add possibility to set temperature to a given operation_mode

* Correctly break

* Review changes

Commit:
Update flux led library

fabaff committed:
Migrate to voluptuous (#3679)

fabaff committed:
Fix sentence (#3709)

mezz64 committed:
Increase allowable polling intensity values (#3711)

fabaff committed:
Adjust vol to accept filenames (fixes #3701) (#3707)

fabaff committed:
Change name of Forecast.io platform to Dark Sky (#3698)

* Rename Forecast.io platform to Dark Sky

* Upgrade to python-forecastio to 1.3.5

* Update to reflect name change (Forecast.io -> Dark Sky)

* Rename forecast to darksky

hartmms committed:
attempting to test new code

hartmms committed:
changed exception to AttributeError
fixed test for above

hartmms committed:
fixed lint errors I made in tests....mqtt/test_init.py

hartmms committed:
more lint fixes for my added test

hartmms commented:
The Travis CI build results were interesting.  On my previous commit, they all passed except for the lint check of my new test code.  Once I fixed the whitespace issues with my code, the new build passed the lint check, but failed the py3.5 build.
